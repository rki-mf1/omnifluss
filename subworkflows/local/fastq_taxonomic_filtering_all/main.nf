include { FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS } from '../../nf-core/fastq_extract_kraken_krakentools/main'

workflow FASTQ_TAXONOMIC_FILTERING_ALL {
    take:
    tools
    ch_reads
    ch_db
    val_taxid

    main:
    ch_kraken2_report           = Channel.empty()
    ch_extracted_kraken2_reads  = Channel.empty()
    ch_multiqc_files            = Channel.empty()
    ch_versions                 = Channel.empty()
    
    if (tools.split(',').contains('kraken2')) {
        FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS(
            ch_reads,
            ch_db,
            val_taxid
        )

        ch_kraken2_report = FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS.out.kraken2_report
        ch_extracted_kraken2_reads = FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS.out.extracted_kraken2_reads
        ch_multiqc_files = ch_multiqc_files.mix(FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS.out.multiqc_files)
        ch_versions = ch_versions.mix(FASTQ_EXTRACT_KRAKEN_KRAKENTOOLS.out.versions)
    }

    emit:
    kraken2_report = ch_kraken2_report
    extracted_kraken2_reads = ch_extracted_kraken2_reads

    multiqc_files     = ch_multiqc_files
    versions          = ch_versions

}
