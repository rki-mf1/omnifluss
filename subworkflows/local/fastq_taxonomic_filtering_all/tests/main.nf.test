nextflow_workflow {

    name "Test Subworkflow FASTQ_TAXONOMIC_FILTERING_ALL"
    script "../main.nf"
    // Incude a nextflow.config for the test if needed
    config "subworkflows/local/fastq_taxonomic_filtering_all/tests/nextflow.config"
    workflow "FASTQ_TAXONOMIC_FILTERING_ALL"

    //  Add tags for the subworkflow
    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/fastq_taxonomic_filtering_all"
    tag "fastq_taxonomic_filtering_all"
    // Add tags for all modules used within this subworkflow.
    tag "kraken2_kraken2"
    tag "kraken2_krakentools"

    // Test name preferably indicating the test-data and file-format used
    test("sarscov2 - single_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = "kraken2"
                input[1] = [ [ id:'test', single_end:true ], // meta map
                            [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true) ]
                        ]
                input[2] = file("$projectDir/assets/dummy_file.txt", checkIfExists: true)
                input[3] = '11308 11320 11520'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - paired_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = "kraken2"
                input[1] = [ [ id:'test', single_end:false ], // meta map
                            [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true) ]
                        ]
                input[2] = file("$projectDir/assets/dummy_file.txt", checkIfExists: true)
                input[3] = '11308 11320 11520'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}