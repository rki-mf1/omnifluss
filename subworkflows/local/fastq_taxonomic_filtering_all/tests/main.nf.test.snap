{
    "sarscov2 - single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.extracted_kraken2_read.fastq.gz:md5,f4296dfe8313535ec8a6700073b96e27"
                    ]
                ],
                "2": [
                    [
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ],
                "extracted_kraken2_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.extracted_kraken2_read.fastq.gz:md5,f4296dfe8313535ec8a6700073b96e27"
                    ]
                ],
                "kraken2_report": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "multiqc_files": [
                    [
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-08-08T13:46:55.992208269"
    },
    "sarscov2 - single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ],
                "extracted_kraken2_reads": [
                    
                ],
                "kraken2_report": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc_files": [
                    "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-08-08T13:47:41.945035683"
    },
    "sarscov2 - paired_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.extracted_kraken2_read_1.fastq.gz:md5,f4296dfe8313535ec8a6700073b96e27",
                            "test.extracted_kraken2_read_2.fastq.gz:md5,2126281657f12a19d697ac5940403e7e"
                        ]
                    ]
                ],
                "2": [
                    [
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ],
                "extracted_kraken2_reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.extracted_kraken2_read_1.fastq.gz:md5,f4296dfe8313535ec8a6700073b96e27",
                            "test.extracted_kraken2_read_2.fastq.gz:md5,2126281657f12a19d697ac5940403e7e"
                        ]
                    ]
                ],
                "kraken2_report": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "multiqc_files": [
                    [
                        "test.kraken2.report.txt:md5,4227755fe40478b8d7dc8634b489761e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-08-08T13:47:21.312294192"
    },
    "sarscov2 - paired_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ],
                "extracted_kraken2_reads": [
                    
                ],
                "kraken2_report": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc_files": [
                    "test.kraken2.report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,16f74c5631271ee9a681a9d291f04f2a",
                    "versions.yml:md5,87b75feb64bd48191f09637f3f1ed463"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-08-08T13:48:02.581921751"
    }
}